bin/spotxBidAdapter.js

```diff
@@ -20,28 +20,42 @@ export const spec = {
    * @return boolean True if this is a valid bid, and false otherwise.
    */
   isBidRequestValid: function(bid) {
-    if (bid && (typeof bid.params !== 'object' || typeof bid.params.video !== 'object')) {
-      utils.logMessage(BIDDER_CODE + ': video params is missing or is incorrect');
+    if (bid && typeof bid.params !== 'object') {
+      utils.logError(BIDDER_CODE + ': params is not defined or is incorrect in the bidder settings.');
       return false;
     }
 
-    if (bid.params.video.channel_id === undefined || bid.params.video.slot === undefined || bid.params.video.video_slot === undefined) {
-      utils.logMessage(BIDDER_CODE + ': channel_id, slot and/or video_slot are not present in bidder params');
+    if (!utils.deepAccess(bid, 'mediaTypes.video')) {
+      utils.logError(BIDDER_CODE + ': mediaTypes.video is not present in the bidder settings.');
       return false;
     }
 
-    const videoSlotDivId = utils.getBidIdParameter('video_slot', bid.params.video);
-    if (window.document.getElementById(videoSlotDivId) === null) {
-      utils.logMessage(BIDDER_CODE + ': video_slot HTML node id does not exist on the page');
+    const playerSize = utils.deepAccess(bid, 'mediaTypes.video.playerSize');
+    if (!playerSize || !utils.isArray(playerSize)) {
+      utils.logError(BIDDER_CODE + ': mediaTypes.video.playerSize is not defined in the bidder settings.');
       return false;
     }
 
-    const slotDivId = utils.getBidIdParameter('slot', bid.params.video);
-    if (window.document.getElementById(slotDivId) === null) {
-      utils.logMessage(BIDDER_CODE + ': slot HTML node id does not exist on the page');
+    if (!utils.getBidIdParameter('channel_id', bid.params)) {
+      utils.logError(BIDDER_CODE + ': channel_id is not present in bidder params');
       return false;
     }
 
+    const videoMediaType = utils.deepAccess(bid, 'mediaTypes.video');
+    const context = utils.deepAccess(bid, 'mediaTypes.video.context');
+    if ((bid.mediaType === 'video' || (videoMediaType && context !== 'outstream')) || utils.getBidIdParameter('ad_unit', bid.params) == 'outstream') {
+      if (!utils.getBidIdParameter('outstream_function', bid.params)) {
+        if (!utils.getBidIdParameter('outstream_options', bid.params)) {
+          utils.logError(BIDDER_CODE + ': please define outstream_options parameter or override the default SpotX outstream rendering by defining your own Outstream function using field outstream_function.');
+          return false;
+        }
+        if (!utils.getBidIdParameter('video_slot', bid.params.outstream_options) || !utils.getBidIdParameter('content_width', bid.params.outstream_options) || !utils.getBidIdParameter('content_height', bid.params.outstream_options)) {
+          utils.logError(BIDDER_CODE + ': please define parameters video_slot, content_width and content_height for object outstream_options in the configuration.');
```

Comment:
Should read "please define parameters video_slot or content_width and content_height"

Also, as a side note, I think we need slot, not necessarily video_slot